Opencv cv::dnn::readNetFromONNX throws error

OpenCV => 4.2.0
Operating System / Platform => Ubuntu 18.04
Compiler => gcc
Detailed description
When I load the onnx model (converted from pytorch ) using **cv::dnn::readNetFromONNX**, it throws error:

```
terminate called after throwing an instance of 'cv::Exception'
  what():  OpenCV(4.2.0) /home/anilsathyan7/opencv/modules/dnn/src/onnx/onnx_importer.cpp:110: error: (-215:Assertion failed) !tensor_proto.raw_data().empty() || !tensor_proto.float_data().empty() || !tensor_proto.double_data().empty() || !tensor_proto.int64_data().empty() in function 'getMatFromTensor'

Aborted (core dumped)
```
 However i'am able to run the onnx models using   **onnxruntime inference session and pytorch** without any problems. Also i was able to load the [mobilenetv2](https://github.com/onnx/models/blob/master/vision/classification/mobilenet/model/mobilenetv2-7.onnx) model using the same setup.

Sample Code:-

```
    //Parameters
    String model = "SINet2.onnx";
    int backendId = 0; // Auto
    int targetId = 0;  // CPU
    int inpWidth = 320;
    int inpHeight = 320;

    // Read network
    CV_Assert(!model.empty());

    Net net = readNetFromONNX(model);//readNet(model, config, framework);
    net.setPreferableBackend(backendId);
    net.setPreferableTarget(targetId);
```


Models: 
[SINet.zip](https://github.com/opencv/opencv/files/5403304/SINet.zip)

It looks like a problem with node **ReduceMax** . In opencv 4.5 i got the following error:-

```
[ERROR:0] global /home/anilsathyan7/opencv-4.5.0/modules/dnn/src/onnx/onnx_importer.cpp (1788) handleNode DNN/ONNX: ERROR during processing node with 1 inputs and 1 outputs: [ReduceMax]:(757)
terminate called after throwing an instance of 'cv::Exception'
  what():  OpenCV(4.5.0) /home/anilsathyan7/opencv-4.5.0/modules/dnn/src/onnx/onnx_importer.cpp:1797: error: (-2:Unspecified error) in function 'handleNode'
> Node [ReduceMax]:(757) parse error: OpenCV(4.5.0) /home/anilsathyan7/opencv-4.5.0/modules/dnn/src/onnx/onnx_importer.cpp:556: error: (-215:Assertion failed) axis >= 2 + i in function 'handleNode'
> 
Aborted (core dumped)

```

--- a/modules/dnn/src/onnx/onnx_importer.cpp
+++ b/modules/dnn/src/onnx/onnx_importer.cpp
@@ -551,11 +551,36 @@ void ONNXImporter::handleNode(const opencv_onnx::NodeProto& node_proto_)
 
                     CV_Assert(axes.size() <= inpShape.size() - 2);
                     std::vector<int> kernel_size(inpShape.size() - 2, 1);
-                    for (int i = 0; i < axes.size(); i++) {
-                        int axis = clamp(axes.get<int>(i), inpShape.size());
-                        CV_Assert_N(axis >= 2 + i, axis < inpShape.size());
-                        kernel_size[axis - 2] = inpShape[axis];
+                    if (axes.size() == 1 && (clamp(axes.get<int>(0), inpShape.size()) <= 1))
+                    {
+                        int axis = clamp(axes.get<int>(0), inpShape.size());
+                        MatShape newShape = inpShape;
+                        newShape[axis + 1] = total(newShape, axis + 1);
+                        newShape.resize(axis + 2);
+                        newShape.insert(newShape.begin(), 2 - axis, 1);
+
+                        LayerParams reshapeLp;
+                        reshapeLp.type = "Reshape";
+                        reshapeLp.name = layerParams.name + "/reshape";
+                        CV_Assert(layer_id.find(reshapeLp.name) == layer_id.end());
+                        reshapeLp.set("dim", DictValue::arrayInt(&newShape[0], newShape.size()));
+
+                        node_proto.set_output(0, reshapeLp.name);
+                        addLayer(reshapeLp, node_proto);
+
+                        kernel_size.resize(2);
+                        kernel_size[0] = inpShape[axis];
+                        node_proto.set_input(0, node_proto.output(0));
                     }
+                    else
+                    {
+                        for (int i = 0; i < axes.size(); i++) {
+                            int axis = clamp(axes.get<int>(i), inpShape.size());
+                            CV_Assert_N(axis >= 2 + i, axis < inpShape.size());
+                            kernel_size[axis - 2] = inpShape[axis];
+                        }
+                    }
+
                     LayerParams poolLp = layerParams;
                     poolLp.name = layerParams.name + "/avg";
                     CV_Assert(layer_id.find(poolLp.name) == layer_id.end());

--- a/modules/dnn/test/test_onnx_importer.cpp
+++ b/modules/dnn/test/test_onnx_importer.cpp
@@ -267,9 +267,11 @@ TEST_P(Test_ONNX_layers, ReduceSum)
     testONNXModels("reduce_sum");
 }
 
-TEST_P(Test_ONNX_layers, ReduceMaxGlobal)
+TEST_P(Test_ONNX_layers, ReduceMax)
 {
     testONNXModels("reduce_max");
+    testONNXModels("reduce_max_axis_0");
+    testONNXModels("reduce_max_axis_1");
 }
 
 TEST_P(Test_ONNX_layers, Scale)

